pkg/misc/io.go

```diff
@@ -0,0 +1,21 @@
+package misc
+
+import "io"
+import "context"
+
+type readerFunc func(p []byte) (n int, err error)
+
+func (rf readerFunc) Read(p []byte) (n int, err error) { return rf(p) }
```

Comment:
Makes me sad that we have to reinvent the wheel here. So at what point will you become a contributor to the golang standard library? xD